app/controllers/teacher_interface/uploads_controller.rb

```diff
@@ -5,10 +5,12 @@ class UploadsController < BaseController
     include ActiveStorage::Streaming
     include HandleApplicationFormSection
     include HistoryTrackable
+    include StreamedResponseAuthenticatable
+
+    skip_before_action :authenticate_teacher!, only: :show
+    before_action -> { authenticate_or_redirect(:teacher) }, only: :show
 
     before_action :redirect_unless_draft_or_further_information
-    before_action :load_application_form
-    before_action :load_document
```

Comment:
Did we mean to remove these lines? I think we use them in the other views.

Reply:
Oops no, amended.